alm/src/utils/contract.ts

```diff
@@ -59,40 +68,43 @@ export const getRequiredSignatures = async (
   return parseInt(requiredSignatures)
 }
 
-export const getValidatorList = async (contract: Contract, blockNumber: number, snapshotProvider: SnapshotProvider) => {
+export const getValidatorList = async (
+  contract: Contract,
+  blockNumber: number,
+  fromHome: boolean,
+  snapshotProvider: SnapshotProvider
+) => {
   const addedEventsFromSnapshot = snapshotProvider.validatorAddedEvents(blockNumber)
   const removedEventsFromSnapshot = snapshotProvider.validatorRemovedEvents(blockNumber)
   const snapshotBlockNumber = snapshotProvider.snapshotBlockNumber()
 
   const fromBlock = snapshotBlockNumber > blockNumber ? snapshotBlockNumber + 1 : blockNumber
   const [currentList, added, removed] = await Promise.all([
     contract.methods.validatorList().call(),
-    contract.getPastEvents('ValidatorAdded', {
-      fromBlock
-    }),
-    contract.getPastEvents('ValidatorRemoved', {
-      fromBlock
-    })
+    getEvents(contract, 'ValidatorAdded', fromBlock, fromHome),
+    getEvents(contract, 'ValidatorRemoved', fromBlock, fromHome)
   ])
 
-  // Ordered desc
-  const orderedEvents = [...addedEventsFromSnapshot, ...added, ...removedEventsFromSnapshot, ...removed].sort(
-    ({ blockNumber: prev }, { blockNumber: next }) => next - prev
-  )
-
-  // Stored as a Set to avoid duplicates
-  const validatorList = new Set(currentList)
-
-  orderedEvents.forEach(e => {
-    const { validator } = e.returnValues
-    if (e.event === 'ValidatorRemoved') {
-      validatorList.add(validator)
-    } else if (e.event === 'ValidatorAdded') {
-      validatorList.delete(validator)
-    }
-  })
-
-  return Array.from(validatorList)
+  // TODO: sort events
```

Comment:
please delete the commented code

Reply:
https://github.com/fuseio/tokenbridge/commit/06772c77f97c4220e5c42a0d5d88ca703b162d18